src/adapter/src/coord/peek.rs

```diff
@@ -261,6 +297,34 @@ pub fn create_fast_path_plan<T: Timestamp>(
                             )));
                         }
                     }
+                    // If there is no arrangement, consider peeking the persist shard directly
+                    let safe_mfp = mfp
+                        .clone()
+                        .into_plan()
+                        .map_err(|e| AdapterError::Unstructured(::anyhow::anyhow!(e)))?
+                        .into_nontemporal()
+                        .map_err(|_e| {
+                            AdapterError::Unstructured(::anyhow::anyhow!(
+                                "OneShot plan has temporal constraints"
+                            ))
```

Comment:
This is a copy-paste of some existing mfp-to-safemfp conversions in the file, including `permute_oneshot_mfp_around_index` which is invoked just below. I don't have any additional context.

Maybe I should factor this out to a function so that it's easier for someone to add a more structured error in the future?

Reply:
> Maybe I should factor this out to a function so that it's easier for someone to add a more structured error in the future?

I think that's fine for now.